tests/devices/qutrit_mixed/test_qutrit_mixed_tracking.py

```diff
@@ -0,0 +1,265 @@
+    (
+        [
+            qml.expval(qml.prod(qml.GellMann(0, 1), qml.GellMann(1, 4))),
+            qml.expval(qml.prod(qml.GellMann(1, 4), qml.GellMann(2, 7))),
+        ],
+        2,
+        20,
+    ),
+    # computational basis measurements
+    ([qml.sample(wires=(0, 1))], 1, 10),
+    ([qml.sample(wires=(0, 1)), qml.expval(qml.GellMann(0, 1))], 2, 20),
+]
+
+
+@pytest.mark.parametrize("mps, expected_exec, expected_shots", shot_testing_combos)
+def test_single_expval(mps, expected_exec, expected_shots):
+    """Test tracker tracks default qutrit mixed execute number of shots for single measurements"""
+    dev = qml.device("default.qutrit.mixed")
+    tape = qml.tape.QuantumScript([], mps, shots=10)
+
+    with dev.tracker:
+        dev.execute(tape)
+
+    assert dev.tracker.totals["executions"] == expected_exec
+    assert dev.tracker.totals["simulations"] == 1
+    assert dev.tracker.totals["shots"] == expected_shots
+
+    tape = qml.tape.QuantumScript([qml.TRX((1.2, 2.3, 3.4), wires=0)], mps, shots=10)
+
+    with dev.tracker:
+        dev.execute(tape)
+
+    assert dev.tracker.totals["executions"] == 3 * expected_exec
+    assert dev.tracker.totals["simulations"] == 1
+    assert dev.tracker.totals["shots"] == 3 * expected_shots
+
+
+@pytest.mark.usefixtures("use_legacy_and_new_opmath")
+def test_multiple_expval_with_prods():
+    """
+    Test tracker tracks default qutrit mixed execute number of shots for new and old opmath tensors.
+    """
+    mps, expected_exec, expected_shots = (
+        [qml.expval(qml.GellMann(0, 1)), qml.expval(qml.GellMann(0, 1) @ qml.GellMann(1, 5))],
+        2,
+        20,
+    )
+    dev = qml.device("default.qutrit.mixed")
+    tape = qml.tape.QuantumScript([], mps, shots=10)
+
+    with dev.tracker:
+        dev.execute(tape)
+
+    assert dev.tracker.totals["executions"] == expected_exec
+    assert dev.tracker.totals["simulations"] == 1
+    assert dev.tracker.totals["shots"] == expected_shots
+
+
+@pytest.mark.logging
+def test_execution_debugging(caplog):
+    """Test logging of QNode forward pass from default qutrit mixed."""
+
+    qml.logging.enable_logging()
+
+    pl_logger = logging.root.manager.loggerDict["pennylane"]
+    plqn_logger = logging.root.manager.loggerDict["pennylane.qnode"]
+
+    # Ensure logs messages are propagated for pytest capture
+    pl_logger.propagate = True
+    plqn_logger.propagate = True
+
+    with caplog.at_level(logging.DEBUG):
+        dev = qml.device("default.qutrit.mixed", wires=2)
+        params = qml.numpy.array(0.1234)
+
+        @qml.qnode(dev, diff_method=None)
+        def circuit(params):
+            qml.TRX(params, wires=0)
+            return qml.expval(qml.GellMann(0, 3))
+
+        circuit(params)
+
+    assert len(caplog.records) == 3
+
+    log_records_expected = [
+        (
+            "pennylane.workflow.qnode",
+            ["Creating QNode(func=<function test_execution_debugging"],
```

Comment:
```suggestion
            ["Creating QNode(func=<function test_execution_debugging>)"],
```
Not sure about this but there is a missing ) and > here

Reply:
It's supposed to be like this, the actual message includes the pointer to the function `Creating QNode(func=<function test_execution_debugging.<locals>.circuit at 0x7c2ed9c63c40> ...` Along with some other stuff.